HIP/hip-0000-token-update-remove-keys.md

```diff
@@ -0,0 +1,196 @@
+   >
+   > - Choose a low-entropy Ed25519 public key; for example, 32 bytes of binary zeros or hex abcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcd.
+   > - In a single TokenUpdate, replace both the admin and wipe keys of the NFT collection with this public key.
+   > - Now to wipe an NFT in this collection, you would have to invert the SHA-512 hash function, which is...going to take a while.
+
+Neither of these approaches is ideal and could easily be solved by allowing the keys to be wiped as part of a `TokenUpdateTransaction` call.
+
+## User stories
+
+- As a creator I want to remove the Wipe Key on my existing NFT collection so that collectors can be assured their NFT can't be removed from their account.
+- As a creator I want to remove the Admin Key on my existing NFT collection so that I can be sure my NFT is immutable.
+- As a creator I want the flexibility to remove keys as my project evolves. For example, I might start out with KYC as a requirement and later decide that it is not necessary.
+- As an NFT minting service I want to be able to mint an NFT collection on behalf of a creator using our private key and then update the treasury account to the creator's account whilst simultaneously removing the admin key so the creator ends up with an immutable NFT collection in their treasury account.
+- As an author of a smart contract that I am now happy is operating smoothly, I would like to remove the admin key
+
+## Specification
+
+1. Introduce a constant to represent an empty key. This will be used as part of the token and/or contract update calls to remove existing keys.
+
+2. Update all other areas of the SDK that make use of an empty key to use this new constant for consistency.
+
+We use the JS SDK for demonstration using the HTS token entity as an example. The same approaches could be take for the `ContractUpdateTransaction`.
+
+### Extend `TokenUpdateTransaction` to Support Removing Keys
+
+All these keys should be removable as part of a `TokenUpdateTransaction` if they are present on a Token:
+
+- Admin Key
+- Wipe Key
+- KYC Key
+- Freeze Key
+- Pause Key
+- Supply Key
+- Fee Schedule Key
+
+**An example `TokenUpdateTransaction` to remove all keys from a Token.**
+
+```js
+let transaction = new TokenUpdateTransaction({
+  tokenId: "0.0.123456",
+  kycKey: Key.None,
+  freezeKey: Key.None,
+  pauseKey: Key.None,
+  wipeKey: Key.None,
+  supplyKey: Key.None,
+  feeScheduleKey: Key.None,
+  adminKey: Key.None,
+}).freezeWithClient(client);
+
+// Sign the transaction with the admin key
+const signTx = await transaction.sign(adminKey);
+```
+
+**Key.None**
+We can’t use `null` as the value for these fields because behind the scenes protobuf removes them for optimisation. An empty KeyList could be used however, since this action can't be undone, we propose having a dedicated constant `Key.None`. The constant provides clearer intent which is crucial when dealing with a transaction that can't be reversed.
```

Comment:
In discussion with some members of my team, we had another thought. The problem is this is a "forever" change. As such, having something really explicit (even more explicit than `Key.None`) would be a good thing. On the other hand, adding a new Key type is ugly because what does it mean to have a Threshold key with a keylist where one of three keys are Key.None? Probably an error?

What if, instead, we just used empty keylist as proposed, but we also add a new boolean to update transactions that is like "-f" and can be used to mean "yes, I really really really want to set a key to null, so do allow this". That would give us the safety of something like Key.None but in a (mostly) compatible way and without having to change the Key type.